Actually functionalize torch.export

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #101433

I thought i enabled this, but apparently not. This PR makes the export fully functional for real this time :)

tugsbayasgalan committed:
Actually functionalize torch.export

[ghstack-poisoned]

gmagogsfm commented on test/export/test_export.py:
```diff
@@ -553,6 +528,25 @@ def fn_ddo(x):
+
+        view_count = 0
+        for node in gm.graph.nodes:
+            if node.op == "call_function" and node.target == torch.ops.aten.add_.Tensor:
+                # No more inplace mutation
+                self.assertTrue(node.target != torch.ops.aten.add_.Tensor)
+            if node.op == "call_function" and node.target == torch.ops.aten.view.default:
+                view_count += 1
```
```suggestion
            if node.op == "call_function":
                # No more inplace mutation
                self.assertNotEqual(node.target, torch.ops.aten.add_.Tensor, "Found unexpected inplace mutation op _add")
                
                if node.target == torch.ops.aten.view.default:
                    view_count += 1
```

gmagogsfm commented on test/export/test_export.py:
```diff
@@ -67,20 +66,9 @@ def forward(self, x):
-            return cond(torch.tensor(x.shape[0] > 4), true_fn, false_fn, [x])
-
 
+@unittest.skipIf(not torchdynamo.is_dynamo_supported(), "dynamo doesn't support")
```
```suggestion
@unittest.skipIf(not torchdynamo.is_dynamo_supported(), "dynamo isn't supported")
```

gmagogsfm commented on test/export/test_passes.py:
```diff
@@ -234,6 +234,18 @@ def forward(self, x):
+
+        ep = export(foo, (x,)).transform(ReplaceViewOpsWithViewCopyOpsPass())
+        # After this pass, there shouldn't be any view nodes in the graph
+        self.assertTrue(count_call_function(ep.module.graph, torch.ops.aten.view_copy.default) > 0)
```
Can you check that there are zero view nodes instead of checking the existence of view_copy nodes?

tugsbayasgalan committed:
Update on "Actually functionalize torch.export"


I thought i enabled this, but apparently not. This PR makes the export fully functional for real this time :) 


[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Actually functionalize torch.export"


I thought i enabled this, but apparently not. This PR makes the export fully functional for real this time :) 


[ghstack-poisoned]

tugsbayasgalan commented:
@pytorchbot merge -f "Unrelated test failures"

PaliC commented:
@pytorchbot revert -c "ignoredsignal" -m "causing failures on functorch macOS tests"

PaliC commented:
Actually this is breaking ` test_python_dispatch.py::TestCustomOp::test_not_implemented_error` as well

tugsbayasgalan commented:
@pytorchbot rebase

tugsbayasgalan committed:
Update on "Actually functionalize torch.export"


I thought i enabled this, but apparently not. This PR makes the export fully functional for real this time :) 


[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Actually functionalize torch.export"


I thought i enabled this, but apparently not. This PR makes the export fully functional for real this time :) 


[ghstack-poisoned]

tugsbayasgalan commented:
@pytorchbot merge
